src/Analyzers/CSharp/Analyzers/SimplifyLinqExpressions/CSharpSimplifyLinqExpressionsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+            var queryNamedType = context.Compilation.GetTypeByMetadataName("System.Linq.Queryable");
+            var enumMethods = enumNamedType?.GetMembers("Where").OfType<IMethodSymbol>();
+            var queryMethods = queryNamedType?.GetMembers("Where").OfType<IMethodSymbol>();
+            AddIfNotNull(whereMethods, enumMethods);
+            AddIfNotNull(whereMethods, queryMethods);
+
+            // add all valid linq calls
+            foreach (var id in validLinqCalls)
+            {
+                enumMethods = enumNamedType?.GetMembers(id).OfType<IMethodSymbol>();
+                queryMethods = queryNamedType?.GetMembers(id).OfType<IMethodSymbol>();
+                AddIfNotNull(linqMethods, enumMethods);
+                AddIfNotNull(linqMethods, queryMethods);
+            }
+
+            if (whereMethods.Count > 0 && linqMethods.Count > 0)
+            {
+                context.RegisterOperationAction(
+                    context => AnalyzeAction(
+                        context, whereMethods.ToImmutable(), linqMethods.ToImmutable()), OperationKind.Invocation);
+            }
+
+            static void AddIfNotNull(ImmutableHashSet<IMethodSymbol>.Builder set, IEnumerable<IMethodSymbol>? others)
+            {
+                if (others != null)
+                {
+                    set.UnionWith(others);
+                }
+            }
+        }
+
+        public void AnalyzeAction(OperationAnalysisContext context, ImmutableHashSet<IMethodSymbol> whereMethods, ImmutableHashSet<IMethodSymbol> linqMethods)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            var semanticModel = context.Operation.SemanticModel;
+            // Check that .Where(...) is not user defined
+            var child = invocationOperation.Children.FirstOrDefault(c => c is IArgumentOperation);
+            var whereClause = child?.Children.FirstOrDefault(c => c is IInvocationOperation);
+            var isLambda = whereClause?.Children.Any(c => c.Syntax.IsKind(SyntaxKind.Argument) &&
+                                                    ((ArgumentSyntax)c.Syntax).Expression.IsKind(SyntaxKind.SimpleLambdaExpression)) ?? false;
+
+            if (whereClause == null ||
+                !(whereClause is IInvocationOperation method) ||
+                method.TargetMethod.OriginalDefinition == null ||
+                !isLambda ||
+                !whereMethods.Contains(method.TargetMethod.OriginalDefinition))
+            {
+                return;
+            }
+
+            // check that the Where clause is followed by a call to a valid method i.e. one of First, FirstOrDefault, Single, SingleOrDefault, etc..
+            // and that it is also not user defined
+            var originalDefinition = (invocationOperation.TargetMethod.ReducedFrom ?? invocationOperation.TargetMethod).OriginalDefinition;
```

Comment:
```suggestion
            var invocationDefinition = invocationOperation.TargetMethod.OriginalDefinition;
```

Reply:
this is also very repetitive.  you twice grab the invocation's target method's original (i.e. you do it in the if-statement above).